lib/attack_test.go

```diff
@@ -266,3 +267,73 @@ func TestMaxBody(t *testing.T) {
 		})
 	}
 }
+
+func TestClient(t *testing.T) {
+	t.Parallel()
+
+	dialer := &net.Dialer{
+		LocalAddr: &net.TCPAddr{IP: DefaultLocalAddr.IP, Zone: DefaultLocalAddr.Zone},
+		KeepAlive: 30 * time.Second,
+		Timeout:   DefaultTimeout,
+	}
+
+	jar, _ := cookiejar.New(nil)
+
+	client := http.Client{
+		Jar: jar,
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			Dial:  dialer.Dial,
+			ResponseHeaderTimeout: DefaultTimeout,
+			TLSClientConfig:       DefaultTLSConfig,
+			TLSHandshakeTimeout:   10 * time.Second,
+			MaxIdleConnsPerHost:   DefaultConnections,
+		},
+	}
+
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}),
+	)
+	defer server.Close()
+
+	tr := NewStaticTargeter(Target{Method: "GET", URL: server.URL})
+	rate := Rate{Freq: 100, Per: time.Second}
+
+	atk := NewAttacker(Client(client))
+	var hits uint64
+	for range atk.Attack(tr, rate, 1*time.Second, "") {
+		hits++
+	}
+
+	if got, want := hits, uint64(rate.Freq); got != want {
```

Comment:
This doesn't actually test that a specific client was used. Perhaps an easy way to do so would be to set a timeout of zero and check for the particular error is returned.